tests/mdev_configuration_allocation_test.go

```diff
@@ -271,4 +273,149 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
+			ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+			var latestPod k8sv1.Pod
+			err := wait.PollImmediate(time.Second, 2*time.Minute, waitForPod(&latestPod, ThisPod(testPod)))
+			return err
+		}
+
+		BeforeEach(func() {
+			Skip("Unbinding older NVIDIA GPUs, such as the Tesla T4 found on vgpu lanes, doesn't work reliably")
+			nodes := libnode.GetAllSchedulableNodes(virtClient).Items
+			Expect(nodes).To(HaveLen(1))
+			node = nodes[0].Name
+			rootPCIId = "none"
+		})
+
+		AfterEach(func() {
+			if CurrentSpecReport().Failed() && rootPCIId != "none" && driverPath != "none" {
+				// The last test went far enough to un-bind the device and then failed.
+				// Make sure we don't leave the device in an unbound state
+				_ = runBashCmdRw(fmt.Sprintf(bindCmdFmt, rootPCIId, driverPath))
+			}
+		})
+
+		It("should create mdevs on devices that appear after CR configuration", func() {
+			By("looking for an mdev-compatible PCI device")
+			out, e, err := runBashCmd(findMdevCapableDevices)
+			Expect(err).ToNot(HaveOccurred(), e)
+			Expect(out).To(ContainSubstring(mdevBusPath))
+			pciId := "'" + filepath.Base(out) + "'"
+
+			By("finding the driver")
+			driverPath, e, err = runBashCmd("readlink -e " + mdevBusPath + pciId + "/driver")
+			Expect(err).ToNot(HaveOccurred(), e)
+			Expect(driverPath).To(ContainSubstring("drivers"))
+
+			By("finding a supported type")
+			out, e, err = runBashCmd(fmt.Sprintf(findSupportedTypeFmt, pciId))
+			Expect(err).ToNot(HaveOccurred(), e)
+			Expect(out).ToNot(BeEmpty())
+			mdevType := filepath.Base(out)
+
+			By("finding the name of the device")
+			fileName := fmt.Sprintf(deviceNameFmt, pciId, mdevType)
+			deviceName, e, err := runBashCmd("cat " + fileName)
+			Expect(err).ToNot(HaveOccurred(), e)
+			Expect(deviceName).ToNot(BeEmpty())
+
+			By("unbinding the device from its driver")
+			re := regexp.MustCompile(`[\da-f]{2}\.[\da-f]'$`)
+			rootPCIId = re.ReplaceAllString(pciId, "00.0'")
+			err = runBashCmdRw(fmt.Sprintf(unbindCmdFmt, rootPCIId, driverPath))
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() error {
+				_, _, err = runBashCmd("ls " + mdevBusPath + pciId)
+				return err
+			}).Should(HaveOccurred(), "failed to disable the VFs on "+rootPCIId)
+
+			By("adding the device to the KubeVirt CR")
+			resourceName := filepath.Base(driverPath) + ".com/" + strings.ReplaceAll(deviceName, " ", "_")
+			kv := util.GetCurrentKv(virtClient)
+			config := kv.Spec.Configuration
+			config.DeveloperConfiguration.FeatureGates = append(config.DeveloperConfiguration.FeatureGates, virtconfig.GPUGate)
+			config.MediatedDevicesConfiguration = &v1.MediatedDevicesConfiguration{
+				MediatedDevicesTypes: []string{mdevType},
+			}
+			config.PermittedHostDevices = &v1.PermittedHostDevices{
+				MediatedDevices: []v1.MediatedHostDevice{
+					{
+						MDEVNameSelector: deviceName,
+						ResourceName:     resourceName,
+					},
+				},
+			}
+			tests.UpdateKubeVirtConfigValueAndWait(config)
```

Comment:
nit: It would be great to have this wrapped in a function, that can be used also at line 135.

Reply:
Thanks for the suggestion but I would still have to pass the `MediatedDevicesTypes` and `MediatedDevices` to the function, since they differ between both cases. Feels that it would be more confusing than anything else...